### PR TITLE
Expanded tags definitions

### PR DESCRIPTION
**Pull request by mscottco**
TLDR; This pull request proposes a specification for expanding the tags options for the devices listed in this database

----

I submitted a [reddit post](https://www.reddit.com/r/Esphome/comments/12n0884/list_of_esphome_products/) looking for a sort of database of devices that were compatible with ESPHome. I've had a go at a modified spec for adding some extended attributes to the devices pages. In essence, it entails:

- board: the board type used in the project. At the moment I've just added esp8266 and esp32. Not sure if it needs to be more granular than that, open to feedback...
- project-url: so there's a nice holder to a link for the product or GitHub repo that contains the details about the project
- made-for-esphome: bool to capture if the manufacturer has conformed to the [made for esphome](https://esphome.io/guides/made_for_esphome.html) spec
- difficulty: to provide a rating for whether this is a 'off the shelf' compatibility. Its based on another commenters categorisation, it seemed a good starting point but happy for feedback.

I spent the day hacking around with gatsby and worked out how to extend the site. Before I put in the legwork to get it working properly, I was hoping you would adopt this as a starting point? From there, I'm happy to build out the devices page template and a couple of list pages to show the data properly.

### CODE REVIEW

**Comment by tekmaven** [comment]
Thanks for the contribution!  These added tags are a good idea, I like the made for esphome tag idea.

**Comment by tekmaven** [comment]
I had to remove the `<br>` tags for now,  it doesn't look as good, but it will comply with the lint policy.

**Comment by mscottco** [comment]
Great, thanks! Sorry about the lint issues, didn't realise it wouldn't pass until it was submitted...

I'll get onto working in some UI modifications to show the new tags and aim for another PR later this week

**Comment by jesserockz** [comment]
@tekmaven please feel free to request my review on any PRs that have the made-for-esphome set to true. We don't have a public list (hoping this site becomes that list) anywhere but I can quickly check for them.
